### PR TITLE
feat: adds an example with macroquad

### PR DESCRIPTION
**Pull request by el-gringo**
I have added an example with macroquad. Which is a library with very few dependencies and compile in few seconds. That fill nicely with a zero dependency like this one, and would allow to test & iterate faster.

I had to reorganize the crate into a workspace in order to not include dependencies of GPUI & Iced.

### CODE REVIEW

**Review by cyypherus — CHANGES_REQUESTED** [review]
Although I do see the drawbacks of linting & loading dev dependencies for examples, I prefer the current structure for the crate so that people can run examples with the example flag. macroquad should simply be added as a [dev dependency](https://doc.rust-lang.org/rust-by-example/testing/dev_dependencies.html)

I would also prefer to recommend people run the new example natively instead of using web - it's easiest & most familiar to simply use `cargo run --example macroquad-example`. Can you remove the web-related files?

Seems like the math module import in the example is unused, can you remove that?

Looks like I have a stray implementation of AnimationTime in my tests that was actually in use - looks like that will have to be deleted since the new one is added to traits.rs.

Thank you for the PR! A macroquad example is a great idea!!

**Comment by cyypherus** [comment]
Oh, about gpui platform support - since windows isn't supported we should keep platform-specific directives. We could add a linux check, or just keep it macos only.

File: examples/macroquad-example/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+use std::f32::consts::PI;
+
+use macroquad::prelude::*;
+use macroquad::math;
```
Comment (cyypherus) [inline examples/macroquad-example/src/main.rs]: Unused import here

File: examples/macroquad-example/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    loop {
+        let time = get_time() as f32 * 1000.;
+        clear_background(LIGHTGRAY);
+
+        let rect_left = animation1.animate_bool(100., screen_width() - 200.0, time);
+        let rotation = animation2.animate_bool(0., 2., time);
+
+        let ox = 300.;
+        let oy = 300.;
```
Comment (cyypherus) [inline examples/macroquad-example/src/main.rs]: What's the o prefix here?

Comment (el-gringo) [reply examples/macroquad-example/src/main.rs]: it is the rotation center, I'll find a more explicit way.

File: examples/macroquad-example/src/main.rs (hunk 1 above)
Comment (el-gringo) [reply examples/macroquad-example/src/main.rs]: ok, I've been a bit fast to do this PR. I'll clean the code

**Comment by el-gringo** [comment]
> Although I do see the drawbacks of linting & loading dev dependencies for examples, I prefer the current structure for the crate so that people can run examples with the example flag. macroquad should simply be added as a [dev dependency](https://doc.rust-lang.org/rust-by-example/testing/dev_dependencies.html)
> 
> I would also prefer to recommend people run the new example natively instead of using web - it's easiest & most familiar to simply use `cargo run --example macroquad-example`. Can you remove the web-related files?
> 
> Seems like the math module import in the example is unused, can you remove that?
> 
> Looks like I have a stray implementation of AnimationTime in my tests that was actually in use - looks like that will have to be deleted since the new one is added to traits.rs.
> 
> Thank you for the PR! A macroquad example is a great idea!!

1. I have not found a way to make the examples work without building all dependencies. That breaks the goal of using macroquad which is a 6s build on my 6yo computer.
2. The web build is optional, the native build works out of the box.

**Comment by cyypherus** [comment]
> I have not found a way to make the examples work without building all dependencies. That breaks the goal of using macroquad which is a 6s build on my 6yo computer.

To workaround this, you can cd into the example directory & `cargo run` from there. :)
It's a little odd that everything is built by default (using `run --example`), but I believe the current crate structure in this repo follows convention.

**Comment by el-gringo** [comment]
Alright, anyway that was not the right place to change the structure.

I've reverted my commit and only pushed my example & cleaned the warnings.

I'll do the `cd example` trick, but my rust-analyzer on my IDE still build iced and use all my RAM.
That's why I am interested in no dependency crates, I have not enough RAM to work on GPUI/Iced :)

**Comment by cyypherus** [comment]
Pulling in your commits in #15, thanks again for the PR!

It's definitely not ideal that cargo / clippy build every example & dependency, I will keep an eye out for any other solutions that still will allow newcomers to use `cargo run --example ...`

btw, it seems like your git configuration does not link your commits to your account! If you figure out what's going on there & would like me to update the metadata on your commits, just let me know.

**Pull request closed without merging**